### PR TITLE
Refactor query engine utilities

### PR DESCRIPTION
**Pull request by Taepper**
### Summary
Just Refactoring query engine utilities:
clearer Operator::negate and Expression::toString, logical Equvalents for debug printing/logging for the Leaf Operators IndexScan and BitmapSelection

## PR Checklist
~~- [X] All necessary documentation has been adapted or there is an issue to do so.~~
~~- [X] The implemented feature is covered by an appropriate test.~~
Just internal refactoring

### CODE REVIEW

**Review by JonasKellerer — APPROVED** [review]
LGTM. Especially removing the database from the toString method.

Only in the the api.cpp, you revert the new poco settings. This should be changed.

File: src/silo_api/api.cpp [hunk 1]
```diff
@@ -245,19 +245,10 @@ class SiloServer : public Poco::Util::ServerApplication {
 
       const silo_api::DatabaseDirectoryWatcher watcher(data_directory, database_mutex);
 
-      auto* const poco_parameter = new Poco::Net::HTTPServerParams;
```
[the remark this answers is not in the document]
Comment (Taepper) [reply src/silo_api/api.cpp:248]: Correct

File: src/silo/query_engine/operators/union.cpp [hunk 2]
```diff
@@ -1,5 +1,6 @@
 #include "silo/query_engine/operators/union.h"
 
+#include <set>
```
[the remark this answers is not in the document]
Comment (Taepper) [reply src/silo/query_engine/operators/union.cpp:3]: Ah, no, only in the follow-up PR

File: include/silo/query_engine/filter_expressions/expression.h [hunk 3]
```diff
@@ -1,10 +1,15 @@
 #pragma once
 
 #include <memory>
+#include <optional>
```
[the remark this answers is not in the document]
Comment (Taepper) [reply include/silo/query_engine/filter_expressions/expression.h:4]: Didn't clean up the imports when seperating the PR in 2 :O

**Pull request merged**